### ISSUE TITLE
Hi, could I ask you a question again?

### ISSUE THREAD

**Issue by Veronica123123** [PTMZ/IMP_GCN#3]
Recently, I try to understand and implement the IMP and LightGCN model, but I have some difficulties. I would be appreciated that if you could help me.
As can be seen in the pictures, there is a difference at the propagation rule between the LightGCN and IMP.  Specifically, in every layer, LightGCN uses "all_emb"(a.k.a., the "E0" ), but in every layer, IMP uses the the embedding of last layer.
I am so confused.
Could you help me understand it?

![capture-2022-04-07-22-18-54](https://user-images.githubusercontent.com/50046260/162221462-ed68fa62-0a84-4d94-bcf7-129daf8f60e4.jpg)
![capture-2022-04-07-22-20-40](https://user-images.githubusercontent.com/50046260/162221479-c554bc0b-87fa-425e-b255-91cc672edbbc.jpg)

**Comment by PTMZ** [comment]
Hi,
Apologies it seems like I made an error for the implementation of LightGCN.
I have updated the code with the following:

```
all_emb_list = [all_emb]
for layer in range(self.n_layers):
    all_emb = torch.sparse.mm(g_droped, all_emb)
    all_emb_list.append(all_emb)
```
Following the LightGCN paper:
https://arxiv.org/pdf/2002.02126.pdf
![image](https://user-images.githubusercontent.com/1089480/162232080-da401204-45f6-4318-a2ce-c55cb0892586.png)
Where N_u and N_i is the neighborhood of the user and item respectively.

Similarly following the IMP-GCN paper:
https://arxiv.org/pdf/2102.10044.pdf
![image](https://user-images.githubusercontent.com/1089480/162233709-b260a4da-d997-4f38-802d-9b00d2f71e7e.png)
Except this time we need to also iterate through the different subgraphs s for each group.

If there are still issues understanding, I can try to explain a bit more in detail.
